datafusion/proto/src/physical_plan/from_proto.rs

```diff
@@ -123,25 +125,39 @@ pub fn parse_physical_sort_exprs(
 ///
 /// # Arguments
 ///
-/// * `proto` - Input proto with physical window exprression node.
+/// * `proto` - Input proto with physical window expression node.
 /// * `name` - Name of the window expression.
-/// * `registry` - A registry knows how to build logical expressions out of user-defined function' names
+/// * `registry` - A registry knows how to build logical expressions out of user-defined function names
 /// * `input_schema` - The Arrow schema for the input, used for determining expression data types
 ///                    when performing type coercion.
 pub fn parse_physical_window_expr(
     proto: &protobuf::PhysicalWindowExprNode,
     registry: &dyn FunctionRegistry,
     input_schema: &Schema,
 ) -> Result<Arc<dyn WindowExpr>> {
-    let codec = DefaultPhysicalExtensionCodec {};
+    parse_physical_window_expr_ext(
+        proto,
+        registry,
+        input_schema,
+        &DefaultPhysicalExtensionCodec {},
+    )
+}
+
+// TODO: Make this the public function on next major release.
```

Comment:
I think what is on main will effectively become the next major release (`38.0.0`) so there is no reason to avoid API changes here.

If we want to backport this to a stable `37.1.0` release (e.g. #9904 ) then we should avoid API changes

Reply:
Ok, I see so if we want the backwards compatible changes it would be against `branch-37`

Reply:
I added the breaking changes here in [1f0040c](https://github.com/apache/arrow-datafusion/pull/10075/commits/1f0040c5d205f6e0cf56aa20cd61f3713b37f4d4)